scripts/Muon/GUI/ElementalAnalysis2/context/ea_group_context.py

```diff
@@ -0,0 +1,109 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from Muon.GUI.ElementalAnalysis2.ea_group import EAGroup
+from mantidqt.utils.observer_pattern import Observable
+
+
+def get_default_grouping(loadedData):
+    '''this creates the first set of groups listed in the grouping table of the Elemental Analysis GUI '''
+    groups = []
+    run_list = loadedData.get_parameter("run")
+    for run_item in run_list:
+        for workspace in loadedData.get_data(run=run_item)["workspace"]:
+            group_name = str(workspace)
+            '''For single workspace names the detector is found by taking everything after ; in the name
+            For co-added workspaces the detector is found by taking everything before '''
+            detector_name = (group_name.split(';', 1)[-1].lstrip()).split('_', 1)[0]
```

Comment:
can we have an example string to hep make sense of these numbers?

Reply:
So an example string for a standard workspace might be '2695; Detector 1' . Everything before the ; (and including the ;) is stripped away to just leave 'Detector 1' .

Co-added workspaces have a different naming convention and so everything after the '_' is removed (I think that the text is missing that information at present). So an example of a Co-added workspace name is 'Detector 1_2695-2686. Everything after the '_' is removed to get 'Detector 1'.